pkg/api/types_test.go

```diff
@@ -2737,3 +2737,45 @@ func TestIsFeatureEnabled(t *testing.T) {
 		})
 	}
 }
+
+func TestKubernetesConfig_GetUserAssignedID(t *testing.T) {
+	k := KubernetesConfig{
+		UseManagedIdentity: true,
+		UserAssignedID:     "fooID",
+	}
+	expected := "fooID"
+
+	if k.GetUserAssignedID() != expected {
+		t.Errorf("expected user assigned ID to be %s, but got %s", expected, k.GetUserAssignedID())
+	}
+
+	k = KubernetesConfig{
+		UseManagedIdentity: false,
+		UserAssignedID:     "fooID",
+	}
+
+	if k.GetUserAssignedID() != "" {
+		t.Errorf("expected user assigned ID to be empty when useManagedIdentity is set to false")
```

Comment:
Should we file a backlog issue to add validation for these pathological scenarios?

Reply:
Validation is already present. This test case just encompasses the behaviour of the method that we had just migrated in this PR

Reply:
cool, thanks for confirming!